pkg/safepath/safepath.go

```diff
@@ -386,12 +387,16 @@ func UnlinkAtNoFollow(path *Path) error {
 		return err
 	}
 	defer fd.Close()
+
+	options := 0
 	if info.IsDir() {
 		// if dir is empty we can delete it with AT_REMOVEDIR
-		return unlinkat(fd.fd, basename, unix.AT_REMOVEDIR)
-	} else {
-		return unlinkat(fd.fd, basename, 0)
+		options = unix.AT_REMOVEDIR
 	}
+	if err = unlinkat(fd.fd, basename, options); err != nil {
+		return fmt.Errorf("failed unlinking path %s: %v", path.String(), err)
```

Comment:
There should be no need to explicitly call `String` with `%v` or `%q` in all the places.

Reply:
Updated, thanks for the review